### PR TITLE
Added appointments and user management

### PR DESCRIPTION
**Pull request by Chirila-Tudor**
The changes here add appointments and user management

### CODE REVIEW

File: src/main/java/ro/chirila/programarispital/exception/FieldValidationException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package ro.chirila.programarispital.exception;
+
+public class FieldValidationException extends RuntimeException{
```
Comment (AlexDorobantiu) [inline src/main/java/ro/chirila/programarispital/exception/FieldValidationException.java]: format your code

File: src/main/java/ro/chirila/programarispital/repository/dto/AppointmentRequestDTO.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+import ro.chirila.programarispital.repository.entity.PeriodOfAppointment;
+import ro.chirila.programarispital.repository.entity.TypeOfService;
+import ro.chirila.programarispital.repository.entity.User;
+
+import java.time.LocalTime;
+import java.util.Date;
+import java.util.List;
+
+public class AppointmentRequestDTO {
```
Comment (AlexDorobantiu) [inline src/main/java/ro/chirila/programarispital/repository/dto/AppointmentRequestDTO.java]: I recommend using the `record` classes from java. They are created especially for DTOs

File: src/main/java/ro/chirila/programarispital/repository/dto/TypeOfServiceDTO.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+public class TypeOfServiceDTO {
+    private String service;
+
+    //region Getter and setter
+    public String getService() {
+        return service;
+    }
+
+    public void setService(String service) {
```
Comment (AlexDorobantiu) [inline src/main/java/ro/chirila/programarispital/repository/dto/TypeOfServiceDTO.java]: you don't need setters, create a constructor with the required fields. Using `record` classes automatically does this for you.

File: src/main/java/ro/chirila/programarispital/repository/dto/UserResponseDTO.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import ro.chirila.programarispital.repository.entity.Role;
+import ro.chirila.programarispital.repository.entity.TypeOfService;
+
+import java.util.List;
+
+public class UserResponseDTO {
+    private String username;
+    private Role role;
+    private List<Appointment> appointments;
+    private List<TypeOfService> services;
```
Comment (AlexDorobantiu) [inline src/main/java/ro/chirila/programarispital/repository/dto/UserResponseDTO.java]: everything here is private

File: src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            TypeOfService typeOfService = typeOfServiceRepository.findByService(modelMapper.map(typeOfServiceDTO, TypeOfService.class).getService());
+
+            if(typeOfService == null){
+                savedAppointment.getTypeOfServices().add(modelMapper.map(typeOfServiceDTO, TypeOfService.class));
+            }else{
+                savedAppointment.getTypeOfServices().add(typeOfService);
+            }
+        }
+
+        if(user.getAppointments() == null){
```
Comment (AlexDorobantiu) [inline src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java]: make the field initialized in the entity

Comment (AlexDorobantiu) [reply src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java]: solved by me

File: src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        }
+        user.getAppointments().add(savedAppointment);
+
+        AppointmentResponseDTO responseDTO = modelMapper.map(appointmentRepository.save(savedAppointment), AppointmentResponseDTO.class);
+        responseDTO.setScheduledPerson(user);
+        return responseDTO;
+
+    }
+
+    public void validateFields(AppointmentRequestDTO appointment){
```
Comment (AlexDorobantiu) [inline src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java]: I recommend doing validation using bean validation like this: https://www.baeldung.com/spring-boot-bean-validation

**Review by AlexDorobantiu — CHANGES_REQUESTED** [review]
See the comments

File: src/main/java/ro/chirila/programarispital/exception/FieldValidationException.java (hunk 1 above)
Comment (Chirila-Tudor) [reply src/main/java/ro/chirila/programarispital/exception/FieldValidationException.java]: fixed(I hope)

File: src/main/java/ro/chirila/programarispital/repository/dto/AppointmentRequestDTO.java (hunk 2 above)
Comment (Chirila-Tudor) [reply src/main/java/ro/chirila/programarispital/repository/dto/AppointmentRequestDTO.java]: Kinda fixed, didn't understand how to access my setter for AppointmentResponseDTO

File: src/main/java/ro/chirila/programarispital/repository/dto/TypeOfServiceDTO.java (hunk 3 above)
Comment (Chirila-Tudor) [reply src/main/java/ro/chirila/programarispital/repository/dto/TypeOfServiceDTO.java]: Fixed

File: src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java (hunk 6 above)
Comment (Chirila-Tudor) [reply src/main/java/ro/chirila/programarispital/service/implementation/AppointmentServiceImpl.java]: Fixed

**Review by AlexDorobantiu — APPROVED** [review]

**Pull request merged**